mrp_bom_location/report/bom_structure.py

```diff
@@ -0,0 +1,64 @@
+# © 2017 Eficent Business and IT Consulting Services S.L.
+#        (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class BomStructureReport(models.AbstractModel):
+    _name = 'report.mrp.mrp_bom_structure_report'
+
+    @staticmethod
+    def _get_child_vals(record, level, qty, uom):
+        child = {
+            'pname': record.product_id.name_get()[0][1],
+            'pcode': record.product_id.default_code,
+            'puom': record.product_uom_id,
+            'uname': record.product_uom_id.name,
+            'level': level,
+            'code': record.bom_id.code,
+            'location_name': record.location_id.complete_name or '',
+        }
+        qty_per_bom = record.bom_id.product_qty
+        if uom:
+            if uom != record.bom_id.product_uom_id:
+                qty = uom._compute_quantity(qty, record.bom_id.product_uom_id)
+            child['pqty'] = (record.product_qty * qty) / qty_per_bom
+        else:
+            # for the first case, the ponderation is right
+            child['pqty'] = (record.product_qty * qty)
+        return child
+
+    def get_children(self, records, level=0):
```

Comment:
You are completely rewritten this method from upstream, so you are destroying inheritability. Please don't do this and try to reimplemented calling super and adding later data.

Reply:
the problem here is that I really need to inherit the inner recursive method. But as the original author didn't think about extensibility of his awesome code there -> the only way I see is to reimplement it to add a single field.